firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -535,4 +553,65 @@ private void redirectToPlayForAabUpdate(String downloadUrl)
     updateAppTaskCompletionSource.setResult(updateState);
     this.updateTask.updateProgress(updateState);
   }
+
+  private void showUpdateDialogIfNewReleaseAvailable() {
+    checkForUpdate()
+        .addOnSuccessListener(
+            latestRelease -> {
+              if (latestRelease != null) {
+                showUpdateAlertDialog(latestRelease);
+              } else {
+                updateToLatestReleaseTaskCompletionSource.setResult(null);
+              }
+            })
+        .addOnFailureListener(
+            e ->
+                setUpdateToLatestReleaseTaskCompletionError(
+                    new FirebaseAppDistributionException(
```

Comment:
If checkForUpdate fails with a FirebaseAppDistributionException, can we set the exception on updateToLatestRelease to the same exception, instead of overwriting the exception?

Reply:
oops added the same comment above :) Probably worth doing the same thing for sign in.